src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/Kepler/COE.cpp

```diff
@@ -641,6 +641,130 @@ COE COE::FromSIVector(const Vector6d& aCOEVector, const AnomalyType& anAnomalyTy
     };
 }
 
+COE COE::FrozenOrbit(
+    const Length& aSemiMajorAxis,
+    const Length& anEquatorialRadius,
+    const Real& aJ2,
+    const Real& aJ3,
+    const Real& anEccentricity,
+    const Angle& anInclination,
+    const Angle& aRaan,
+    const Angle& anAop,
+    const Angle& aTrueAnomaly
+)
+{
+    if (!aSemiMajorAxis.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Semi-major Axis");
+    }
+    if (!anEquatorialRadius.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Equatorial Radius");
+    }
+    if (!aJ2.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("J2");
+    }
+    if (!aJ3.isDefined() || aJ3 == 0.0)
+    {
+        throw ostk::core::error::runtime::Undefined("J3");
+    }
+    if (!aRaan.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Raan");
+    }
+    if (!aTrueAnomaly.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("True anomaly");
+    }
+
+    // TBI: block inclination + eccentricity both defined
+
+    const Angle critical_inclinations[] = {Angle::Degrees(63.4349), Angle::Degrees(116.5651)};
+    const Angle critical_aops[] = {Angle::Degrees(90.0), Angle::Degrees(270.0)};
+
+    const Real re = anEquatorialRadius.inMeters();
+    const Real j2 = aJ2;
+    const Real j3 = aJ3;
+
+    // ecc =~ ecc_coefficient * sin(incl)
+    const Real ecc_coefficient = -j3 * re / 2.0 / j2 / aSemiMajorAxis.inMeters();
```

Comment:
```suggestion
    const Real eccCoefficient = -j3 * equatorialRadius_meters / 2.0 / j2 / aSemiMajorAxis.inMeters();
```
We camel case in the C++ code